datafusion/common/src/scalar.rs

```diff
@@ -1330,8 +1330,10 @@ impl ScalarValue {
                         Some(scalar_vec)
                     }
                 };
-                let data_type = nested_type.data_type().clone();
-                ScalarValue::List(value, Box::new(data_type))
+                ScalarValue::List(
+                    value,
+                    Box::new(Field::new("item", nested_type.data_type().clone(), true)),
```

Comment:
this is effectively going to lose the original field name (which isn't attached to the `array`) in this particular codepath, which is a similar problem to what I observed in https://github.com/apache/arrow-datafusion/issues/2874

However, this PR certainly seems better than the current state of master which loses the field name through all paths